client/lib/cart/store/cart-synchronizer.js

```diff
@@ -106,17 +107,27 @@ CartSynchronizer.prototype.update = function( changeFunction ) {
 		// with a consistent state when we apply our local changes. The strategy
 		// here is to queue the changes until the first request has completed, then
 		// proceed as normal.
-		this._enqueueChange( changeFunction );
+		this._enqueueChange( changeFunction, serverFlushCallback );
+		if ( ! this._activeRequest ) {
+			this.fetch();
+		}
+		debug( 'update: change enqueued' );
 		return;
 	}
 
+	debug( 'update: processing change' );
+	this._addFlushCallback( serverFlushCallback );
+	this._processChangeFunction( changeFunction );
+	this.emit( 'change' );
+};
+
+CartSynchronizer.prototype._processChangeFunction = function ( changeFunction ) {
 	if ( this._activeRequest && this._activeRequest.state !== 'canceled' ) {
 		this._activeRequest.state = 'canceled';
 	}
 
 	this._latestValue = changeFunction( this._latestValue );
 	this._performRequest( 'update', this._postToServer.bind( this ) );
-	this.emit( 'change' );
```

Comment:
That is something we can do in addition, but we still need the callback thing to integrate with the signup framework IMO, to hook it [here](https://github.com/Automattic/wp-calypso/pull/11401/files/0d0fcb1dbbfe1363a0ba97c3fdb4e7b2c85e444d#diff-160d380c201b4db2d947c495e877a98aR44), I'm not sure how would we do it with the even you're suggesting.